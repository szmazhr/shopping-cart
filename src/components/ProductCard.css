.product-card{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 300px;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.5);
  transition: box-shadow 0.3s ease-in-out;
}
.product-card:hover{
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.product-card .product-card-info{
  padding: 0 1rem;
  width: 100%;
  text-align: center;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.product-card .product-card-info a{
  color: rgb(27, 27, 27);
  text-decoration: none;
}
.product-card .product-card-info a:hover{
  text-decoration: underline;
}

.product-card .product-card-footer{
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  position: relative;
}
.product-card .product-card-price{
  font-size: 1.2rem;
  font-weight: bold;
}
.product-card .product-card-btn{
  background-color: rgb(164, 255, 225);
  padding: 0.4rem .8rem;
  border-radius: 5px;
  font-size: 1.1rem;
  cursor: pointer;
}
.product-card .product-card-btn:hover{
  background-color: rgb(130, 255, 213);
}
.product-card .product-card-img{
  display: flex;
  max-height: 200px;
}
.product-card .product-card-img img{
  width: 100%;
  object-fit: cover;
}