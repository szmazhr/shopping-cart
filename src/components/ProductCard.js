import { Link } from 'react-router-dom';
import './ProductCard.css';

function ProductCard({product, addToCart}) {

  const buttonClickHandler = (e) => {
    e.target.textContent = 'Added';
    addToCart(product, 1);
  }

  return (
    <div className='product-card'>
      <div className='product-card-img'>
        <img src={product.image} loading="lazy"  alt={product.title}/>
      </div>
      <div className='product-card-info'>
        <Link to={`/product/${product.id}`}>
          <h3 className='product-card-title'>{product.title}</h3>
        </Link>
        <div className='product-card-footer'>
        <p className='product-card-price'>${product.price}</p>
        <button className='product-card-btn' onClick={buttonClickHandler}>Add to Cart</button>
        </div>
      </div>
    </div>
  )
}

export default ProductCard