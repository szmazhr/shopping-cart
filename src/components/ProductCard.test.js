import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import ProductCard from "./ProductCard";

describe("ProductCard", () => {
  const product = {
    id: 10,
    title: "Product 1",
    image: "https://via.placeholder.com/300",
    price: 10,
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ProductCard product={product} addToCart={() => {}} />, { wrapper: MemoryRouter }
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls addToCart when add to cart button is clicked', () => {
    const addToCart = jest.fn();
    render(
      <ProductCard product={product} addToCart={addToCart} />, { wrapper: MemoryRouter }
    );
    userEvent.click(screen.getByText('Add to Cart'));
    expect(addToCart).toHaveBeenCalledTimes(1);
    expect(addToCart).toHaveBeenCalledWith(product, 1);
  });


});