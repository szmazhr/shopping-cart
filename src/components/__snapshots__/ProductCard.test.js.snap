// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card-img"
    >
      <img
        alt="Product 1"
        loading="lazy"
        src="https://via.placeholder.com/300"
      />
    </div>
    <div
      class="product-card-info"
    >
      <a
        href="/product/10"
      >
        <h3
          class="product-card-title"
        >
          Product 1
        </h3>
      </a>
      <div
        class="product-card-footer"
      >
        <p
          class="product-card-price"
        >
          $10
        </p>
        <button
          class="product-card-btn"
        >
          Add to Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
